lib/cloud-watch-event-listener.js

```diff
@@ -168,14 +159,41 @@ class CloudWatchEventListener extends events.EventEmitter {
         metadataSource: 'db',
       };
       
+      // Note that since we're now inserting instances directly into the
+      // database now, we're always going to have a pending state in the db for
+      // managed instances.  This means that most 'pending' state messages will
+      // be ignored.
+      if (dbResponse.lastEvent && dbResponse.lastEvent === generated) {
+        log.warn('whoa!');
+      } 
+
       if (dbResponse.lastEvent && dbResponse.lastEvent < generated) {
         try {
-          if (state === 'pending' || state === 'running') {
-            await this.state.updateInstanceState({region, id, state, lastEvent: generated}, transaction);
-            log.info(logInfo, 'CloudWatch Event resulting in insertion');
-          } else {
+          // We only want to handle the terminated case specially, since in
+          // other cases we'll filter for only pending and running as query
+          // options when talking to the db
+          if (state === 'terminated') {
+            await this.state.upsertTermination({
+              id: dbResponse.id,
+              workerType: dbResponse.workerType,
+              region: dbResponse.region,
+              az: dbResponse.az,
+              instanceType: dbResponse.instanceType,
+              imageId: dbResponse.instanceType,
+              launched: dbResponse.launched,
+              terminated: generated,
+              lastEvent: generated,
```

Comment:
I know `generated` isn't a new term here, but I keep finding it confusing as to what it is. Maybe indicate that it is a date or something?

Reply:
it's the Date that the message from cloudwatch was generated.  I could rename it, but then we'd have a really long date.  There is a comment explaining that, yes, the name sucks, but explains what it is.
```js
    // Not a great name, but this variable is the time that the CloudWatch
    // Event message was generated by the API, not the time at which we
    // received it
    let generated = new Date(body.time);
```